### PR TITLE
Setup Prisma Client along with Postgres connection

### LINKED ISSUE #1: Setup Prisma Client with Postgres Connection
## Description
This issue is created to track the task of setting up Prisma Client within our project and establishing a connection to a Postgres database. Prisma Client will be used for database access and querying within the application.

## Tasks
- [x] Add Prisma as a dependency in the `package.json` file.
- [x] Configure the `schema.prisma` file with the necessary settings for connecting to the Postgres database.
- [x] Create a `prisma` directory to store Prisma-related files.
- [x] Implement a sample database query using Prisma in a test file to ensure proper setup.

## Additional Information
(Provide any additional context, details, or specific requirements related to this task.)

## Acceptance Criteria
- Prisma Client is successfully set up and configured.
- The application can establish a connection to a Postgres database using Prisma.
- A sample database query using Prisma retrieves the expected results.

## Definition of Done
- [x] All tasks mentioned above are completed.
- [x] The changes have been tested locally and verified.
- [x] Documentation, if necessary, has been updated.

## Related Issues
(If there are any related issues, mention them here.)

### PR DESCRIPTION
**Pull request by Tamrakar182**
## Description
This pull request addresses the task of setting up Prisma Client and establishing a connection to a Postgres database. Prisma Client will be utilized for database access and querying within the application.

## Changes Made
- [x] Added Prisma as a dependency in the `package.json` file.
- [x] Configured the `schema.prisma` file with the necessary settings for connecting to the Postgres database.
- [x] Created a `prisma` directory to store Prisma-related files.
- [x] Implemented a sample database query using Prisma in a test file to ensure proper setup.

## Testing
- [x] Tested the Prisma Client setup locally with a local Postgres database.
- [x] Verified that the sample database query is successful and retrieves the expected results.

## Screenshots
(Optional: Include screenshots or code snippets if applicable.)
- User Schema
image

- DB data creation and query
image

## Related Issues
Closes #1 

## Checklist
- [x] I have tested these changes locally.
- [x] I have updated the documentation if necessary.
- [x] My code follows the coding standards of the project.
- [x] I have performed a self-review of my code.
- [x] I have added appropriate comments to my code, especially in areas that may be unclear.
- [x] I have squashed/rebased my commits.

## Additional Notes
(Add any additional information or context that may be helpful for reviewers.)

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id        Int      @id @default(autoincrement())
```
Comment (subhashishjs) [inline prisma/schema.prisma]: use uuid for primary key

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
```
Comment (subhashishjs) [inline prisma/schema.prisma:10]: create a .env.example file

**Review by subhashishjs — CHANGES_REQUESTED** [review]
Work on folder structure. follow nestJS folder structure from our backend repo. There should be a src folder where we have index.js or server.js file which is entry point for our database. I recommend you to follow this folder structure.
https://github.com/rakimsth/fashion-fusion

File: prisma/schema.prisma (hunk 1 above)
Comment (Tamrakar182) [reply prisma/schema.prisma]: updated

File: prisma/schema.prisma (hunk 2 above)
Comment (Tamrakar182) [reply prisma/schema.prisma:10]: updated

**Comment by Tamrakar182** [comment]
i have resolved all the issues regarding the folder structure and added controllers and routes

File: package.json [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
   "version": "1.0.0",
-  "description": "init",
-  "main": "index.js",
+  "description": "Simple Express app with Prisma and PostgreSQL",
+  "main": "src/index.js",
   "type": "module",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "dev": "nodemon index.js"
+    "dev": "nodemon index.js",
```
Comment (subhashishjs) [inline package.json:9]: will this script work?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import express from 'express';
+import dotenv from 'dotenv';
+import indexRouter from './routes/index.js';
+import cors from 'cors';
+
+dotenv.config();
+
+const app = express();
+const PORT = 5000;
```
Comment (subhashishjs) [inline src/index.js]: use .env folder for this purpose

**Review by subhashishjs — APPROVED** [review]
LGTM

File: package.json (hunk 3 above)
Comment (Tamrakar182) [reply package.json:9]: works

File: src/index.js (hunk 4 above)
Comment (Tamrakar182) [reply src/index.js]: fixed

**Pull request merged**